### PR TITLE
Map ccn

### PR DESCRIPTION
**Pull request by danlee1025**
- Added `ccn-mapping.properties` file to map source endpoint of message header to ccn
    - Removed ccn as a constructor arg

### CODE REVIEW

File: src/main/java/org/mitre/hapifhir/MedMorphToCIBMTR.java [hunk 1]
```diff
@@ -24,35 +32,38 @@
 import org.apache.http.impl.client.CloseableHttpClient;
 import org.apache.http.impl.client.HttpClients;
 import org.apache.http.util.EntityUtils;
-import java.util.stream.Collectors;
 
 public class MedMorphToCIBMTR {
   private String cibmtrUrl;
   private String ccn;
+  private String authToken;
```
Comment (dehall) [inline src/main/java/org/mitre/hapifhir/MedMorphToCIBMTR.java]: We should clarify how this class is intended to be used. Do we intend (a) just one instance of MedMorphToCIBMTR where every request invokes `convert` on that instance, or (b) is there one instance of MedMorphToCIBMTR per request? If (a), then we don't want the CCN and auth token to be stored as state - they should be passed to the various functions as needed. This is fine if (b) though. I was originally assuming (a) but based on what's on the `cibmtr` branch on medmorph-fhir-server it seems like we're going with (b). If that's the case can you add some comments saying that's the intended use?

File: src/main/java/org/mitre/hapifhir/MedMorphToCIBMTR.java [hunk 2]
```diff
@@ -197,4 +208,16 @@ protected JSONObject getMeta() {
 
     return metaObject;
   }
+
+  // Returns CCN after mapping message header source
+  protected String getCcn(MessageHeader messageHeader) {
+    try (InputStream input = getClass().getClassLoader().getResourceAsStream("ccn-mapping.properties")) {
```
Comment (dehall) [inline src/main/java/org/mitre/hapifhir/MedMorphToCIBMTR.java:211]: A properties file makes sense for mapping, but I'm not sure how we will be able to configure it once this is plugged into the medmorph-fhir-server. If we put a `ccn-mapping.properties` file somewhere there does this pick it up?

Comment (danlee1025) [reply src/main/java/org/mitre/hapifhir/MedMorphToCIBMTR.java:211]: Hmm should I load the properties file in `medmorph-fhir-server` and then pass it in as arg when we construct `MedMorphToCIBMTR`?

File: src/main/java/org/mitre/hapifhir/MedMorphToCIBMTR.java (hunk 1 above)
Comment (danlee1025) [reply src/main/java/org/mitre/hapifhir/MedMorphToCIBMTR.java]: I think we actually do want (a) where we only have one instance of `MedMorphToCIBMTR` so I'll remove `ccn` and `authToken` as state and change how I instantiate `MedMorphToCIBMTR` on the `cibmtr` branch.

File: src/main/java/org/mitre/hapifhir/MedMorphToCIBMTR.java (hunk 2 above)
Comment (dehall) [reply src/main/java/org/mitre/hapifhir/MedMorphToCIBMTR.java:211]: Yeah that could work. My only concern, and I'm probably overthinking this, is that we may have to restart the server if we change the file. Certainly not the end of the world, especially if the server starts up quickly, but would be preferable if changes to the file were seen "live", ie, without a restart

Comment (danlee1025) [reply src/main/java/org/mitre/hapifhir/MedMorphToCIBMTR.java:211]: > If we put a ccn-mapping.properties file somewhere there does this pick it up?

So I just tested this by removing the properties file and adding it to `medmorph-fhir-server` and it gets picked up.

> but would be preferable if changes to the file were seen "live", ie, without a restart

Is there a setting to make this work? I tried changing the file while the server was running but the changes do not get picked up until I restart the server.

Comment (dehall) [reply src/main/java/org/mitre/hapifhir/MedMorphToCIBMTR.java:211]: If you tried this code as-is and it didn't pick it up file changes, then there's not going to be any trivial way to do it. I guess the classloader or something must hold onto the initial version and not re-fetch it every time it's read.

Let's hold off on that for now, we can discuss the options with CIBMTR next week and figure out the best approach.

**Comment by danlee1025** [comment]
Found a different approach to extract CCN so closing this PR for now.

**Pull request closed without merging**